manipulation/src/main/java/org/jboss/gm/manipulation/actions/PomTransformer.java

```diff
@@ -0,0 +1,91 @@
+package org.jboss.gm.manipulation.actions;
+
+import static org.jboss.gm.common.ProjectVersionFactory.withGAV;
+
+import org.commonjava.maven.atlas.ident.ref.ProjectVersionRef;
+import org.gradle.api.Action;
+import org.gradle.api.XmlProvider;
+import org.jboss.gm.common.alignment.AlignmentModel;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+/**
+ * Overrides data in pom generated by the old 'maven' plugin.
+ */
+public class PomTransformer implements Action<XmlProvider> {
```

Comment:
If a Gradle project generates pom files for deployment, does it retrieve the information from the original unmodified build.gradle or from the in-memory altered model?

Reply:
That depends on how the pom is generated. I think the most common is using "maven" plugin or "maven-publish" plugin. IIRC both of them use unresolved (i.e. the original) dependency versions, so that's why this transformer.

Reply:
When I was trying things I also saw that the pom files did not reflect dependency updated the changes OOTB